### ISSUE TITLE
Platform-based select not working on BUILD file of external repository

### ISSUE THREAD

**Issue by carlosgalvezp** [bazelbuild/bazel#22727]
### Description of the bug:

Hi,

I have a tarball `foo` downloaded via `http_archive` in the `WORKSPACE`, and in the corresponding `BUILD.foo.bazel` I have the following:

```
filegroup(
  name = "all_files",
  srcs = glob(["**"]) + select({
      "@main_workspace//:x86_64_config": ["@x86//:all_files"],
      "@main_workspace//:arm_config": ["@arm//:all_files"],
  }),
)
```

The problem is that even if I pass `--platforms=//:arm` to `bazel build`, the `select` above always picks the `x86` branch, instead of the `arm` branch. This does NOT happen on a regular `BUILD` file in the main repository - the correct branch is taken in that case.

Platforms are defined as:

```
platform(
  name = "arm",
  constraint_values = ["@platforms//cpu:arm"],
)

config_setting(
  name = "arm_config",
  constraint_values = ["@platforms//cpu:arm"],
)
```

When debugging this via `bazel config <hash>` I can clearly see that it's picking the local `host` platform instead of the `arm` platform that I'm requesting via `--platforms`.

Is this expected? Could it be that the platform is set _after_ repository rules (such as `http_archive`) are instantiated?

### Which category does this issue belong to?

Core

### Which operating system are you running Bazel on?

Ubuntu 22.04

### What is the output of `bazel info release`?

release 7.1.2

**Comment by fmeum** [comment]
Could you turn this into a standalone reproducer? I could debug it.

**Comment by carlosgalvezp** [comment]
I've managed to create a minimal repro now, pushed here:

https://github.com/carlosgalvezp/bazel_22727

Command to reproduce:

```
bazel build --config=clang --config=arm  --aspects //tools/bazel/aspect:foo.bzl%foo :foo
```

This leads to:
```
external/clang/BUILD:6:10: configurable attribute "srcs" in @@clang//:all_files doesn't match this configuration. Would a default condition help?

Conditions checked:
 //tools/bazel/platforms:arm_none_config
```

Note: **this behavior does not occur if do not use the aspect**, or if I remove the `foo_wrapper` attribute from the aspect. 

My guess is that when creating the `foo_wrapper` it's targetting an x86 platform (`cfg = "exec"`) and is failing there due to missing default condition.

My real problem is that, if I add a default condition, the aspect is still executed with the x86 files, instead of with the arm files. 

Perhaps what I'm trying to achieve is just not possible, because `all_files` is a static target that first gets resolved to x86? Maybe instead I should do:

```
filegroup(
  name = "all_files_x86",
)

filegroup(
  name = "all_files_arm",
)
```

And then use each filegroup in the respective toolchain?

**Comment by pcjanzen** [comment]
For various reasons, the python rules require a cc_toolchain, so your host toolchain will have to pass analysis (and therefore it will need some default condition), even if it won't ever be used.

It's still not exactly clear what you're really trying to do: the aspect in your reproducer doesn't create any actions and doesn't use any toolchains.

**Comment by carlosgalvezp** [comment]
I see, makes sense. The problem is that it picks x86 files at analysis time, but at execution time I need arm files. Anyhow I solved it by replacing the `select` with 2 different filegroups.

> It's still not exactly clear what you're really trying to do: the aspect in your reproducer doesn't create any actions and doesn't use any toolchains.

Yes, I've intentionally removed as much as possible to create a minimal example of the problem. The real aspect essentially runs a linter on the code, and depends on the C++ toolchain. 

I will close the issue since I found a workaround. If you feel this needs further investigation on your side let me know! Thanks for the help :)
